contributors/devel/sig-node/separate-mount-namespace.md

```diff
@@ -0,0 +1,212 @@
+   container volumeMounts with `mountPropagation: Bidirectional` are visible to
+   each other and container volumeMounts with `MountPropagation:
+   HostToContainer`
+
+The fact that mounts originating in the Container Runtime, OCI hooks, Kubelet,
+and container volumeMounts with `mountPropagation: Bidirectional` were visible
+to the host OS in the original implementations of Kubernetes was accidental to
+the implementation, and there are a number of real-world benefits to changing
+the way Kubernetes is installed in order to hide the container-specific mount
+points from the host OS.
+
+![Fig 2: Container-Runtime-scoped mount propagation](images/hidden-k8s-mount-propagation.png)
+
+_Fig 2: Container-Runtime-scoped mount propagation_
+
+## Benefits
+
+### Systemd Efficiency
+
+Today systemd's CPU usage is adversely affected by a large number of
+mountpoints, especially versions released prior to December 2020. This feature
+of systemd, which scans all /proc/1/mount to invoke udev events for every
+change suffers from two issues. The first was an inefficiency in systemd’s
+processing logic. This was fixed in December 2020, and this fix reportedly
+reduced the systemd mount-scanning-related overhead by 30-50%. The second is a
+lack of granularity in the events the kernel sends to systemd about mountpoint
+changes, which necessitates a full rescan of all mountpoints even on a minor
+change or single mountpoint addition. The kernel change to remedy this seems
+unlikely to move forward in the near term.  (See
+[BZ1819869](https://bugzilla.redhat.com/show_bug.cgi?id=1819868) for more
+details)
+
+For example, a fairly small OpenShift deployment today (which does not yet have
+the systemd December 2020 fix) has upwards of 400 mountpoints on every node
+owned by Kubelet, the Container Runtime, and infrastructure containers like OVN
+or CSI operators. Hiding the container mountpoints reduces the steady-state
+systemd CPU utilization from ~10% to ~0% on every worker node. On a more
+heavily loaded system, with 1000’s of container mountpoints, systemd’s usage
+will cap out at 100% of a single CPU core. Even with the 30-50% reduction in
+systemd usage expected from the December 2020 fix, having systemd take up as
+much as 50-70% of a single CPU in the worst-case scenario is a lot of
+unnecessary overhead.
+
+Administrators of other OSes with other host OS services that monitor or
+process mountpoints indiscriminately could also benefit from hiding these
+container-specific mountpoints from the OS at large.
+
+### Tidiness and Encapsulation
+
+These container overlay mounts, secret and configmap mounts, and namespace
+mounts are all internal implementation details around the way the Kubernetes
+and the Container Runtime allocate resources to containers. For normal
+day-to-day operation, there is no need for these special-use mounts to be
+visible in the host OS's top-level 'mount' or 'findmnt' output.
+
+### Security
```

Comment:
cc @kubernetes/sig-security-pr-reviews

Reply:
Hey, turns out this proposed mechanism likely prevents containers (even privileged containers) from exploiting [CVE-2021-33910](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-33910)!